src/dmd/dtoh.d

```diff
@@ -0,0 +1,1724 @@
+             m.parent && m.parent.ident == DMDModule.dmd &&
+             !m.parent.parent)
+        {
+            return true;
+        }
+        return false;
+    }
+
+    private bool isFrontendModule(ASTCodegen.Module m)
+    {
+        if (!m || !m.parent)
+            return false;
+
+        // Ignore dmd.root
+        if (m.parent.ident == DMDModule.root &&
+            m.parent.parent && m.parent.parent.ident == DMDModule.dmd &&
+            !m.parent.parent.parent)
+        {
+            return false;
+        }
+
+        // Ignore dmd.visitor and derivatives
+        if ((m.ident == DMDModule.visitor ||
+             m.ident == DMDModule.parsetimevisitor ||
+             m.ident == DMDModule.permissivevisitor ||
+             m.ident == DMDModule.strictvisitor ||
+             m.ident == DMDModule.transitivevisitor) &&
+             m.parent && m.parent.ident == DMDModule.dmd &&
+             !m.parent.parent)
+        {
+            return false;
+        }
+        return ((m.parent.ident == DMDModule.dmd && !m.parent.parent) ||
+                (m.parent.parent.ident == DMDModule.dmd && !m.parent.parent.parent));
+    }
+//}
+
+void genCppHdrFiles(ref Modules ms)
+{
+    DMDType._init();
+    //version(BUILD_COMPILER)
+    //{
+        DMDModule._init();
+        DMDClass._init();
+    //}
+
+    OutBuffer buf;
+    buf.writestring("#pragma once\n");
+    buf.writeByte('\n');
+    buf.printf("// Automatically generated by dmd -HC\n");
+    buf.writeByte('\n');
+    buf.writestring("#include <assert.h>\n");
+    buf.writestring("#include <stddef.h>\n");
+    buf.writestring("#include <stdio.h>\n");
+    buf.writestring("#include <string.h>\n");
+    buf.writeByte('\n');
+    buf.writestring("#define _d_void void\n");
+    buf.writestring("#define _d_bool bool\n");
+    buf.writestring("#define _d_byte signed char\n");
+    buf.writestring("#define _d_ubyte unsigned char\n");
+    buf.writestring("#define _d_short short\n");
+    buf.writestring("#define _d_ushort unsigned short\n");
+    buf.writestring("#define _d_int int\n");
+    buf.writestring("#define _d_uint unsigned\n");
+    if (global.params.isLP64)
+    {
+        buf.writestring("#define _d_long long\n");
+        buf.writestring("#define _d_ulong unsigned long\n");
+    }
+    else
+    {
+        buf.writestring("#define _d_long long long\n");
+        buf.writestring("#define _d_ulong unsigned long long\n");
+    }
+    buf.writestring("#define _d_float float\n");
+    buf.writestring("#define _d_double double\n");
+    buf.writestring("#define _d_real long double\n");
+    buf.writestring("#define _d_char char\n");
+    buf.writestring("#define _d_wchar wchar_t\n");
+    buf.writestring("#define _d_dchar unsigned\n");
+    buf.writestring("typedef _d_long d_int64;\n");
+    buf.writestring("\n");
+    buf.writestring("#define _d_null NULL\n");
+    buf.writestring("\n\n");
+
+    OutBuffer check;
+    check.writestring(`
+    #if OFFSETS
```

Comment:
Please put all if this in the debug clock below.

Reply:
`check` is used unconditionally below

Reply:
I meant all the uses of it, everything between line 230-269.

Reply:
That change results in wrong headers for non-debug builds....